cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRuntimeProvider.java

```diff
@@ -85,6 +85,12 @@ public ProgramRunner createProgramRunner(ProgramType type, Mode mode, Injector i
                                 type, ProgramType.SPARK);
 
     switch (mode) {
+      case IN_MEMORY:
```

Comment:
@chtyim I went over a few iterations of how to get an `InMemorySparkProgramRunner` to work with the existing classloader logic with `SparkProgramRunner`. In the end, I decided on this as the best way since I didn't think it was a good idea for `InMemorySparkProgramRunner` to implement a `ProgramClassLoader` since it's in the same "container" as the SparkProgramRunner.

As a result of this, whenever creating a classloader, there is not extra logic to check if the program runner is an `InMemoryProgramRunner`, and if so, extract the underlying program runner.

What do you think about this? Is there a better way?